### PR TITLE
projects: add example project ltc2672

### PR DESCRIPTION
**Pull request by mrcsosa**
example project for hardware checking of ltc2672 on dc2903a

Signed-off-by: Marc Paolo Sosa <marcpaolo.sosa@analog.com>

### CODE REVIEW

File: projects/dc2903a/src/examples/basic_example/basic_example.h [hunk 1]
```diff
@@ -0,0 +1,51 @@
+/***************************************************************************//**
+ *   @file   basic_example.h
+ *   @brief  basic example header for DC2903A project
+ *   @author JSanBuenaventura (jose.sanbuenaventura@analog.com)
+********************************************************************************
+ * Copyright 2022(c) Analog Devices, Inc.
```
Comment (amiclaus) [inline projects/dc2903a/src/examples/basic_example/basic_example.h:6]: nit: 2023

File: projects/dc2903a/src/platform/maxim/main.c [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+	ret = iio_example_main();
+#endif
+	return ret;
+
+#if (BASIC_EXAMPLE + IIO_EXAMPLE == 0)
+#error At least one example has to be selected using y value in Makefile.
+#elif (BASIC_EXAMPLE + IIO_EXAMPLE > 1)
+#error Selected example projects cannot be enabled at the same time. \
+Please enable only one example and re-build the project.
+#endif
```
Comment (amiclaus) [inline projects/dc2903a/src/platform/maxim/main.c:88]: nit: add extra line after #endif

File: projects/dc2903a/src/platform/maxim/parameters.c [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+struct max_uart_init_param ltc2672_uart_extra_ip = {
+	.flow = UART_FLOW_DIS,
+};
+#endif
+
+struct max_spi_init_param ltc2672_spi_extra_ip  = {
+	.numSlaves = 1,
+	.polarity = SPI_SS_POL_LOW,
+	.vssel = MXC_GPIO_VSSEL_VDDIOH,
+};
```
Comment (amiclaus) [inline projects/dc2903a/src/platform/maxim/parameters.c:59]: nit: new line

File: projects/dc2903a/src/platform/platform_includes.h [hunk 4]
```diff
@@ -0,0 +1,53 @@
+/***************************************************************************//**
+ *   @file   platform_includes.h
+ *   @brief  Includes for used platforms used by max11205pmb1 project.
+ *   @author JSanBuenaventura (jose.sanbuenaventura@analog.com)
+********************************************************************************
+ * Copyright 2022(c) Analog Devices, Inc.
```
Comment (amiclaus) [inline projects/dc2903a/src/platform/platform_includes.h:6]: 2023

File: projects/max11205pmb1/Makefile [hunk 5]
```diff
@@ -1,6 +1,6 @@
 # Select the example you want to enable by chossing y for enabling and n for disabling
-BASIC_EXAMPLE=n
-IIO_EXAMPLE=y
+BASIC_EXAMPLE=y
```
Comment (amiclaus) [inline projects/max11205pmb1/Makefile:2]: this changes are unrelated to the dc2903a project. please remove them, and add separate commit if needed.

File: projects/max11205pmb1/src/examples/basic_example/basic_example.c [hunk 6]
```diff
@@ -60,32 +60,32 @@ int basic_example_main()
 	struct max11205_dev *max11205_desc;
 	struct no_os_irq_ctrl_desc *max11205_gpio_irq_desc;
 	int16_t adc_data_raw;
-	int32_t adc_data_mv;
-	bool new_data_avail;
+	int32_t adc_data_mv = 100;
+	bool new_data_avail = true;
```
Comment (amiclaus) [inline projects/max11205pmb1/src/examples/basic_example/basic_example.c:64]: ditto.

**Review by amiclaus — COMMENTED** [review]
looks good overall, some minor comments + some unrelated changes.

File: projects/max11205pmb1/src/examples/basic_example/basic_example.c [hunk 7]
```diff
@@ -66,7 +66,7 @@ int basic_example_main()
 	/* Initialize GPIO IRQ controller */
 	ret = no_os_irq_ctrl_init(&max11205_gpio_irq_desc, &max11205_gpio_irq_ip);
 	if (ret)
-		return ret;
```
Comment (CiprianRegus) [inline projects/max11205pmb1/src/examples/basic_example/basic_example.c:69]: These changes are not needed.
